extensions/ql-vscode/src/databases/github-database-prompt.ts

```diff
@@ -30,47 +30,60 @@ export async function findGitHubDatabasesForRepository(
 }
 
 /**
- * Prompt the user to download a database from GitHub and download that database.
+ * Ask whether the user wants to download a database from GitHub.
+ * @return true if the user wants to download a database, false otherwise.
  */
-export async function promptAndDownloadGitHubDatabase(
-  octokit: Octokit,
-  owner: string,
-  repo: string,
+export async function askForGitHubDatabaseDownload(
   databases: CodeqlDatabase[],
   config: GitHubDatabaseConfig,
-  databaseManager: DatabaseManager,
-  storagePath: string,
-  cliServer: CodeQLCliServer,
-  commandManager: AppCommandManager,
-): Promise<void> {
+): Promise<boolean> {
   const languages = databases.map((database) => database.language);
 
   const message =
     databases.length === 1
       ? `This repository has an origin (GitHub) that has a ${getLanguageDisplayName(
           languages[0],
-        )} CodeQL database. Connect to GitHub and download the existing database?`
+        )} CodeQL database. Download the existing database from GitHub?`
       : `This repository has an origin (GitHub) that has ${joinLanguages(
           languages,
-        )} CodeQL databases. Connect to GitHub and download any existing databases?`;
+        )} CodeQL databases. Download any existing databases from GitHub?`;
 
   const answer = await showNeverAskAgainDialog(
     message,
     false,
-    "Connect",
+    "Download",
     "Not now",
     "Never",
   );
 
   if (answer === "Not now" || answer === undefined) {
-    return;
+    return false;
   }
 
   if (answer === "Never") {
     await config.setDownload("never");
-    return;
+    return false;
   }
 
+  return true;
+}
+
+/**
+ * Download a database from GitHub by asking the user for a language and then
+ * downloading the database for that language.
+ */
+export async function downloadDatabaseFromGitHub(
```

Comment:
Ah now it became a separate function :+1: Should it move to `github-database-module.ts` as it doesnt implement any prompting itself?

Reply:
I'll rename the file to `github-database-download`, then it makes sense to have both of these functions in the same file. This will match with #3079 where I introduce these two functions for updating databases.